Soccar [UI]/src/nl/soccar/ui/fx/drawable/CarUiFx.java

```diff
@@ -118,4 +100,64 @@ private void drawWheel(WheelPhysics wheel, GraphicsContext gc) {
         gc.restore();
     }
 
+    private void drawBoostTrail(List<Point2D> trail, GraphicsContext gc) {
+
+        double prevX;
+        double prevY;
+        double x;
+        double y;
+        double o = 0.0;
+
+        // TODO: Boost color should be the color of the team the player is in.
+        int r = 30;
+        int g = 30;
+        int b = 230;
+
+        // Cancel when there is no trail.
+        if (trail.isEmpty()) {
+            return;
+        }
+
+        // Line settings
+        gc.setLineWidth(DisplayConstants.BOOST_TRAIL_WIDTH);
+        gc.setLineCap(StrokeLineCap.ROUND);
+
+        // Draw trail line
+        for (int i = 1; i < trail.size(); i++) {
+
+            gc.setStroke(Color.rgb(r, g, b, o));
+
+            prevX = PhysicsUtilities.toPixelX((float) trail.get(i - 1).getX());
+            prevY = PhysicsUtilities.toPixelY((float) trail.get(i - 1).getY());
+
+            x = PhysicsUtilities.toPixelX((float) trail.get(i).getX());
+            y = PhysicsUtilities.toPixelY((float) trail.get(i).getY());
+
+            gc.strokeLine(prevX, prevY, x, y);
+            o = Math.min(1.0F, o + (1.0F / PhysicsConstants.CAR_BOOST_TRAIL_SIZE));
+        }
+    }
+
+    @Override
+    public void draw(GraphicsContext context) {
+        CarPhysics physics = super.getPhysicsModel();
+
+        if (ClientController.getInstance().getCurrentPlayer().getUsername().equals(super.getModel().getPlayer().getUsername())) {
+            SteerAction steerAction = Keyboard.getSteerAction();
+            HandbrakeAction handbrakeAction = Keyboard.getHandbrakeAction();
+            ThrottleAction throttleAction = Keyboard.getThrottleAction();
+
+            super.getModel().setSteerAction(steerAction);
+            super.getModel().setHandbrakeAction(handbrakeAction);
+            super.getModel().setThrottleAction(throttleAction);
+
+            // TODO : Change location of sending this message (Ugly implementation for now)

```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1135)